### PR TITLE
Create DefaultCosmosDBServiceFactory with AzureComponentFactory

### PR DESCRIPTION
**Pull request by chiangvincent**
The PR instantiates the `DefaultCosmosDBServiceFactory`, which uses the AzureComponentFactory passed in from the scale controller. Previously, the ICosmosDBServiceFactory was using the DI implementation, which was using the wrong AzureComponentFactory. This generated tokens that did not correctly authorize on the cosmos side.

## Private Stamp Testing Logs 
[[Kusto Link]](https://wawstest.kusto.windows.net/wawskustotest?query=H4sIAAAAAAAEAH2QzWrDMBCE736KJacYEtmSrNgx5FBKDoG6GOwXUNVtq6IfI28bCn342jn01p6WgflmmSkK6C5gYghoyMYAmuBZE5L1uBWlkPuy2ZdHEGUr6raSTHElGl7nWVHAgIGAA36u929Q8rY6sqaWlVzBwWiH9zFQis5hOq%2f0nH3D9Q0Twk32yXqdvgbSfnrUHuF0gs3LRzCT06QDbeDX3yc0dsZxeXuzwxPSFTHA9p8anEmhDqo55MAYKG9DvgROKb4vK8B46c7DeNf1O%2bhwnvUr7uBhKemy7AebZdFsLwEAAA%3d%3d&web=0)

// MI connection at datetime(2023-08-09 20:27:43.5152817)
// Sent 1 event at datetime(2023-08-09 20:31:49.8734317)
ScaleControllerEvents
| where EventPrimaryStampName == "funcplatant" 
| where PreciseTimeStamp between (datetime(2023-08-09 20:27:41.3256586) .. 5min)
| project TIMESTAMP, Message, Level

### CODE REVIEW

**Review by yojagad — APPROVED** [review]

File: src/WebJobs.Extensions.CosmosDB/Trigger/CosmosDbScalerProvider.cs [hunk 1]
```diff
@@ -31,11 +32,12 @@ public CosmosDbScalerProvider(IServiceProvider serviceProvider, TriggerMetadata
                 azureComponentFactory = serviceProvider.GetService<AzureComponentFactory>();
             }
 
+            IConfiguration config = serviceProvider.GetService<IConfiguration>();
             ILoggerFactory loggerFactory = serviceProvider.GetService<ILoggerFactory>();
             CosmosDbMetadata cosmosDbMetadata = JsonConvert.DeserializeObject<CosmosDbMetadata>(triggerMetadata.Metadata.ToString());
             cosmosDbMetadata.ResolveProperties(serviceProvider.GetService<INameResolver>());
             IOptions<CosmosClientOptions> options = serviceProvider.GetService<IOptions<CosmosClientOptions>>();
-            ICosmosDBServiceFactory serviceFactory = serviceProvider.GetService<ICosmosDBServiceFactory>();
+            ICosmosDBServiceFactory serviceFactory = new DefaultCosmosDBServiceFactory(config, azureComponentFactory);
```
Comment (ealsur) [inline src/WebJobs.Extensions.CosmosDB/Trigger/CosmosDbScalerProvider.cs:40]: Are there any tests that need updating?

**Comment by chiangvincent** [comment]
Investigated with @NickCraver and pipeline tests appear to be broken after scaler API PR was merged

File: test/WebJobs.Extensions.CosmosDB.Tests/CosmosDBEndToEndTests.cs [hunk 2]
```diff
@@ -151,6 +151,7 @@ private static class EndToEndTestClass
                 [QueueTrigger("NotUsed")] QueueItem item,
                 [CosmosDB(DatabaseName, CollectionName, Id = "{DocumentId}")] JObject document,
                 [CosmosDB(DatabaseName, CollectionName, SqlQuery = "SELECT * FROM c where c.input = {Input}")] IEnumerable<Item> documents,
+                //[CosmosDB(DatabaseName, CollectionName, SqlQuery = "SELECT * FROM c where c.id = {DocumentId}")] IEnumerable<Item> documents1,
```
Comment (ealsur) [inline test/WebJobs.Extensions.CosmosDB.Tests/CosmosDBEndToEndTests.cs]: Should this be uncommented?

Comment (alrod) [reply test/WebJobs.Extensions.CosmosDB.Tests/CosmosDBEndToEndTests.cs]: removed

**Review by ealsur — APPROVED** [review]

**Pull request merged**